### ISSUE TITLE
user-agent header gets overwritten

### ISSUE THREAD

**Issue by Somebi** [Codeception/Codeception#4576]
Only by changing to User-Agentt header appears inside printed by php headers list.
```
modules:
    config:
        PhpBrowser:
          headers:
            User-Agentt: Mozilla/5.0 (iPhone; CPU iPhone OS 9_1 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13B143 Safari/601.1
```
PHP received headers inside acceptance test:
```
[HTTP_USER_AGENTT] => Mozilla/5.0 (iPhone; CPU iPhone OS 9_1 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13B143 Safari/601.1

[HTTP_USER_AGENT] => Symfony2 BrowserKit
```

This is not working:
```
modules:
    config:
        PhpBrowser:
          headers:
            User-Agent: Mozilla/5.0 (iPhone; CPU iPhone OS 9_1 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13B143 Safari/601.1
```

**Comment by Naktibalda** [comment]
It works if you set `User_Agent` instead of `User-Agent`.

**Comment by Naktibalda** [comment]
The actual result I got was `User-Agent: Mozilla/5.0 (iPhone; CPU iPhone OS 9_1 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13B143 Safari/601.1,Symfony BrowserKit`

**Comment by Naktibalda** [comment]
Fixed.

The root cause was the long chain of libraries: PhpBrowser -> BrowserKit -> Guzzle -> curl.
Headers option was passed to constructor of Guzzle, so BrowserKit was not aware of these headers.
Constructor options are default values for requests in Guzzle, when BrowserKit received request from PhpBrowser, it checked if User-Agent is set, and set it's own if it wasn't.
User-Agent set in request by BrowserKit was considered to be the real one by Guzzle and it was used instead of the "default" one set in constructor parameters.

I fixed the issue by making PhpBrowser to keep headers to itself, this way the headers traverse the whole chain and nothing overrides them.
$headers property was introduced to InnerBrowser a while back by refactoring of haveHttpHeader method.
